napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
+
+    For each channel in a multi-channel image and a multi-channel label array,
+    it measures properties of a reference channel (default: 0) and a probe
+    channel.
+    For each object in the reference channel, it returns descriptive statistics
+    of overlapping objects in the probe channel(s).
+    Objects are considered to overlap when object in probe channel has more
+    than 'intersection_area_over_object_area' of its area inside object in 
+    ref channel (0 = No overlap, 1 = Full overlap, default = 0.5).
+    If single labeled image is given, regular 'regionprops_table' function is
+    executed.
+
+    Parameters
+    ----------
+    labels_array : (C, M, N[,P]) ndarray
+        Multichannel labels array. Channel must be the first dimention.
+    intensity_image : (C, M, N[,P]) ndarray, optional
+        Multichannel image. Channel must be the first dimention.
+    ref_channel : int, optional
+        Reference channel number. Default is 0.
+    intersection_area_over_object_area : float, optional
+        Ratio of area that a probe channel object must overlap with a reference
+        channel object. It can be understood as area of intersection divided by
+        area of probe object. Ranges from 0 to 1. Default is 0.5.
+    summary: bool, optional
+        Determine whether to return summary statistics or direct relatioships.
+        Default is True.
+    suffixes: List[str], optional
+        Suffixes to identify each channel. List must match number of channel in
+        'labels_array'. If None, then '-ch' + number is used.
+
+    Returns
+    -------
+    table_list : List
+        List of tables where each table relates one probe channel objects to
+        the reference channel objects. If `summary` is True, for each property,
+        returns count, mean, standard deviation, median, min, max, 25th
+        quartile and 75th quartile. If `summary` is False, direct relationships
+        are returned (which may lead to columns with repeated labels).
+    """
+    # quantitative analysis using scikit-image's regionprops
+    from skimage.measure import regionprops_table as sk_regionprops_table
+    import numpy as np
+    import pandas as pd
+    # To DO: check if input shape is correct
+    table_list = []
+    # Single channel case (single label image provided)
+    if ref_channel is None:
+        table_list += [regionprops_table(image = intensity_image,
+                                        labels = labels_array,
+                                        **kwargs)]
+        return table_list
+    # Channel axis is expected to be 0
+    n_channels = labels_array.shape[0]
+   
+    large_numbers = False
+    # If not provided, create default suffixes
+    if not suffixes:
+        suffixes = ['-ch' + str(i) for i in range(n_channels)]
+    else:
+        suffixes = ['-' + suffix for suffix in suffixes]
+
+    def highest_overlap(regionmask, intensity_image,
```

Comment:
Could this be a `_highest_overlap` function on top level? It would then also be easier to write a test for it.